dbt/include/sqlserver/macros/adapters.sql

```diff
@@ -107,14 +109,27 @@
   {{ return(load_result('check_schema_exists').table) }}
 {% endmacro %}
 
+
+{% macro sqlserver__create_view_exec(relation, sql) -%}
+    {%- set temp_view_sql = sql.replace("'", "''") -%}
+    execute('create view {{ relation.include(database=False) }} as
+    {{ temp_view_sql }}
+    ');
+{% endmacro %}
+
+
 {% macro sqlserver__create_view_as(relation, sql) -%}
-  create view {{ relation.schema }}.{{ relation.identifier }} as
-    {{ sql }}
+    use [{{ relation.database }}];
+    {{ sqlserver__create_view_exec(relation, sql) }}
 {% endmacro %}
```

Comment:
@semcha was the reason for this to make the macro run in a single transaction?

Reply:
What is your suggestion? :)

Reply:
I think my suggestion would be:
```sql
{% macro sqlserver__create_view_as(relation, sql) -%}
  use [{{ relation.database }}];
  create view {{ relation.include(database=false) }} as
    {{ sql }}
{% endmacro %}
```

Reply:
@swanderz That way you get an error `'CREATE VIEW' must be the first statement in a query batch.`